src/dftbp/derivs/perturb.F90

```diff
@@ -286,7 +286,7 @@ subroutine wrtEField(this, env, parallelKS, filling, eigvals, eigVecsReal, eigVe
     integer, intent(in), allocatable :: nNeighbourCam(:)
 
     !> Charge mixing object
-    type(TMixerReal), intent(inout), allocatable :: pChrgMixer
+    class(TMixerReal), intent(inout), allocatable :: pChrgMixer
```

Comment:
Please, remove the `p`-prefix from the variable name (`pChrgMixer` -> `chrgMixer`). We used that to indicate pointers, but mixer instances are (fortunately) not pointers any more.